middle_end/flambda2/from_lambda/lambda_to_flambda.ml

```diff
@@ -535,32 +538,39 @@ let compile_staticfail acc env ccenv ~(continuation : Continuation.t) ~args :
        allocation region"
       Continuation.print continuation;
   let rec add_end_regions acc ~region_stack_now =
-    (* CR pchambart: this probably can't be exercised right now, no lambda
-       jumping through a region seems to be generated. *)
+    (* This can maybe only be exercised right now using "match with exception",
+       since that causes jumps out of try-regions (but not normal regions). *)
     (* CR pchambart: This closes all the regions between region_stack_now and
        region_stack_at_handler, but closing only the last one should be
        sufficient. *)
-    let add_end_region region ~region_stack_now after_everything =
+    let add_end_region (region : Env.region_stack_element) ~region_stack_now
+        after_everything =
       let add_remaining_end_regions acc =
         add_end_regions acc ~region_stack_now
       in
       let body = add_remaining_end_regions acc after_everything in
       fun acc ccenv ->
-        CC.close_let acc ccenv
-          (Ident.create_local "unit")
-          Not_user_visible Flambda_kind.With_subkind.tagged_immediate
-          (End_region region) ~body
+        match region with
+        | Try_with _ -> body acc ccenv
+        | Regular region_ident ->
+          CC.close_let acc ccenv
+            (Ident.create_local "unit")
+            Not_user_visible Flambda_kind.With_subkind.tagged_immediate
+            (End_region region_ident) ~body
     in
     let no_end_region after_everything = after_everything in
     match
       Env.pop_region region_stack_now, Env.pop_region region_stack_at_handler
     with
     | None, None -> no_end_region
-    | Some (region1, region_stack_now), Some (region2, _) ->
-      if Ident.same region1 region2
+    | ( Some
+          ( ((Regular region_ident1 | Try_with region_ident1) as region1),
+            region_stack_now ),
+        Some ((Regular region_ident2 | Try_with region_ident2), _) ) ->
+      if Ident.same region_ident1 region_ident2
```

Comment:
The pattern-matching seems a bit heavier than necessary. Maybe we could add a new `same_region: Env.region_stack_element -> Env.region_stack_element -> bool` and use that in the condition ?
I can submit a PR if you want.

Reply:
(https://github.com/mshinwell/flambda-backend/pull/24)